mlflow/transformers.py

```diff
@@ -0,0 +1,994 @@
+from mlflow.utils.model_utils import (
+    _validate_and_copy_code_paths,
+    _validate_and_prepare_target_save_path,
+    _download_artifact_from_uri,
+    _get_flavor_configuration_from_uri,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+
+FLAVOR_NAME = "transformers"
+_PIPELINE_BINARY_KEY = "pipeline"
+_PIPELINE_BINARY_FILE_NAME = "pipeline"
+_COMPONENTS_BINARY_KEY = "components"
+_MODEL_KEY = "model"
+_TOKENIZER_KEY = "tokenizer"
+_FEATURE_EXTRACTOR_KEY = "feature_extractor"
+_IMAGE_PROCESSOR_KEY = "image_processor"
+_PROCESSOR_KEY = "processor"
+_TOKENIZER_TYPE_KEY = "tokenizer_type"
+_FEATURE_EXTRACTOR_TYPE_KEY = "feature_extractor_type"
+_IMAGE_PROCESSOR_TYPE_KEY = "image_processor_type"
+_PROCESSOR_TYPE_KEY = "processor_type"
+_CARD_TEXT_FILE_NAME = "model_card.md"
+_CARD_DATA_FILE_NAME = "model_card_data.yaml"
+_TASK_KEY = "task"
+_INSTANCE_TYPE_KEY = "instance_type"
+_PIPELINE_MODEL_TYPE_KEY = "pipeline_model_type"
+_MODEL_PATH_OR_NAME_KEY = "source_model_name"
+_SUPPORTED_SAVE_KEYS = {_MODEL_KEY, _TOKENIZER_KEY, _FEATURE_EXTRACTOR_KEY, _IMAGE_PROCESSOR_KEY}
+_SUPPORTED_RETURN_TYPES = {"pipeline", "components"}
+_logger = logging.getLogger(__name__)
+
+
+def _model_packages(model) -> List[str]:
+    """
+    Determines which pip libraries should be included based on the base model engine
+    type.
+
+    :param model: The model instance to be saved in order to provide the required underlying
+                  deep learning execution framework dependency requirements.
+    :return: A list of strings representing the underlying engine-specific dependencies
+    """
+    engine = _get_engine_type(model)
+    if engine == "torch":
+        return ["torch", "torchvision"]
+    else:
+        return [engine]
+
+
+@experimental
+def get_default_pip_requirements(model) -> List[str]:
+    """
+    :param model: The model instance to be saved in order to provide the required underlying
+                  deep learning execution framework dependency requirements. Note that this must
+                  be the actual model instance and not a Pipeline.
+    :return: A list of default pip requirements for MLflow Models that have been produced with the
+             ``transformers`` flavor. Calls to :py:func:`save_model()` and :py:func:`log_model()`
+             produce a pip environment that contain these requirements at a minimum.
+    """
+
+    from transformers import TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel
+
+    if not isinstance(model, (TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel)):
+        raise MlflowException(
+            "The supplied model type is unsupported. The model must be one of: "
+            "PreTrainedModel, TFPreTrainedModel, or FlaxPreTrainedModel",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    base_reqs = ["transformers"]
+    try:
+        base_reqs.extend(_model_packages(model))
```

Comment:
```suggestion
    try:
        base_reqs = ["transformers", *_model_packages(model)]
```

A minor simplification :)